packages/editor/CodeMirror/CodeMirrorControl.test.ts

```diff
@@ -208,4 +208,43 @@ describe('CodeMirrorControl', () => {
 
 		expect(control.editor.state.doc.toString()).toBe(expected);
 	});
+
+	it('updateBody should update the note ID facet and dispatch changes', () => {
+		const control = createEditorControl('test');
+		const noteIdFacet = control.joplinExtensions.noteIdFacet;
+		const getFacet = () => control.editor.state.facet(noteIdFacet);
+
+		control.updateBody('Test', { noteId: 'updated' });
+		expect(getFacet()).toBe('updated');
+
+		// Updating the ID without updating the body should change the ID
+		control.updateBody('Test', { noteId: 'updated 2' });
+		expect(getFacet()).toBe('updated 2');
+
+		// Changing the body, without specifying a new note ID, should
+		// not update the facet.
+		control.updateBody('Test 2');
+		expect(getFacet()).toBe('updated 2');
+	});
+
+	it('updateBody should dispatch changes to the note ID', () => {
+		const control = createEditorControl('test');
+
+		let noteId = '';
+		const noteIdChangeListener = EditorState.transactionExtender.of(transaction => {
+			for (const effect of transaction.effects) {
+				if (effect.is(control.joplinExtensions.setNoteIdEffect)) {
+					noteId = effect.value;
+				}
+			}
+			return null;
+		});
+		control.addExtension(noteIdChangeListener);
+
+		control.updateBody('Test', { noteId: 'updated' });
+		expect(noteId).toBe('updated');
+
+		control.updateBody('Test', { noteId: 'updated-2' });
+		expect(noteId).toBe('updated-2');
+	});
```

Comment:
These tests demonstrate:
1. Getting the current note ID using `state.facet`.
1. Using a CodeMirror [`transactionExtender`](https://codemirror.net/docs/ref/#state.EditorState%5EtransactionExtender) to watch for changes to the note ID open in the editor.  If the note body changes, the `setNoteIdEffect` is included in the same transaction. One possible use case is determining whether a change to the note body was caused by a user or switching to a new note.